added release notes for 6.0.8-20 (#1135)
RedisLabs/redislabs-docs: content/platforms/release-notes/k8s-6-0-8-20-2020-12.md

---
Title: Redis Enterprise for Kubernetes Release Notes 6.0.8-20 (December 2020)
description: Release notes for version 6.0.8-20 (latest)
weight: 72
alwaysopen: false
categories: ["Platforms"]
---

> The Redis Enterprise K8s 6.0.8-20 release is a *major release* on top of [6.0.8-1](https://github.com/RedisLabs/redis-enterprise-k8s-docs/releases/tag/v6.0.8-1) providing support for the latest [Redis Enterprise Software release 6.0.8-30](https://docs.redislabs.com/latest/rs/release-notes/rs-6-0-8-september-2020/) and includes several enhancements  and bug fixes.

Reviewer: ```suggestion
The Redis Enterprise K8s 6.0.8-20 release is a *major release* on top of [6.0.8-1](https://github.com/RedisLabs/redis-enterprise-k8s-docs/releases/tag/v6.0.8-1) providing support for the latest [Redis Enterprise Software release 6.0.8-30](https://docs.redislabs.com/latest/rs/release-notes/rs-6-0-8-september-2020/) and includes several enhancements and bug fixes.
```

## Overview

This release of the operator provides:

>  * The latest release version of the operator

Reviewer: @alexmilowski It seems redundant to say that this release of the operator provides...the latest release version of the operator. Isn't that a tautology?

Author: Each release is a release and this is the latest.

>  * New features

Reviewer: @alexmilowski Can you be a bit more specific?

Author: They are listed below.

 * Various bug fixes

> Follow the instructions in ["Upgrading a Redis Enterprise Cluster in Operator-based Architecture"]({{< relref "/platforms/kubernetes/upgrading-kubernetes-with-operator.md" >}}) to upgrade to this Kubernetes operator release.

Reviewer: ```suggestion
To upgrade your deployment to this latest release, see ["Upgrading a Redis Enterprise Cluster in Operator-based Architecture"]({{< relref "/platforms/kubernetes/upgrading-kubernetes-with-operator.md" >}}).
```

## Images

> 

Reviewer: ```suggestion
This release includes the following container images:
```

>  * **Redis Enterprise** - redislabs/redis:6.0.8-30 or redislabs/redis:6.0.8-30.rhel7-openshift

Reviewer: ```suggestion
 * **Redis Enterprise**: redislabs/redis:6.0.8-30 or redislabs/redis:6.0.8-30.rhel7-openshift
```

>  * **Operator** - redislabs/operator:6.0.8-20

Reviewer: ```suggestion
 * **Operator**: redislabs/operator:6.0.8-20
```

>  * **Services Rigger** - redislabs/k8s-controller:6.0.8-20 or redislabs/services-manager:6.0.8-20 (Red Hat registry)

Reviewer: ```suggestion
 * **Services Rigger**: redislabs/k8s-controller:6.0.8-20 or redislabs/services-manager:6.0.8-20 (Red Hat registry)
```

> 

Reviewer: @alexmilowski ^^ When naming these images, can you link to them directly on DockerHub?

Author: They are in more than one place.

## New features

>  * Databases custom resources (REDB) can be created in separate consumer namespaces from the operator and cluster namespace. The operator deployment is configured to watch specific namespaces for these database REDB CR’s.

Reviewer: ```suggestion
 * You can now create database custom resources (REDB CRs) in consumer namespaces that are separate from the operator and cluster namespace. You configured the operator deployment to watch specific namespaces for these REDB CRs.
```

Reviewer: @alexmilowski Please write in the active voice. Also, please confirm that I have not changed the meaning here.

 * The [Gesher admission control proxy](https://github.com/RedisLabs/redis-enterprise-k8s-docs/blob/master/admission/GESHER.md) is now certified by Red Hat.

>  * REDB CR’s no longer require a Redis Enterprise cluster name. The name will default to the cluster in the context of the operator.

Reviewer: ```suggestion
 * REDB CRs no longer require a Redis Enterprise cluster name. The name will default to the cluster in the context of the operator.
```

Reviewer: @alexmilowski When you say "the name will default to..." what name are you referring to? Do you mean that the REDB CRs will use the default cluster name?

>  * REC and REDB CR’s are now validated via a schema.

Reviewer: ```suggestion
 * REC and REDB CRs are now validated via a schema.
```

Reviewer: @alexmilowski What does REC stand for? It's not clear.

## Important fixes

>  * Using database controller (REDB) no longer generates errors:  “failed to update database status" (RED44919)

Reviewer: ```suggestion
 * The database controller (REDB) no longer generates the following error message:  “failed to update database status" 
```

 * Issues with configuring replica-of through the database controller (REDB) and TLS have been fixed. (RED48285)

>  * A time out issue with “rlutil upgrade” was fixed. (RED48700)

Reviewer: ```suggestion
 * A timeout issue with `rlutil upgrade` was fixed. (RED48700)
```

## Known limitations

### CrashLoopBackOff causes cluster recovery to be incomplete  (RED33713)

> When a pod status is CrashLoopBackOff and we run the cluster recovery, the process will not complete. The workaround is to delete the crashing pods manually and recovery process will continue.

Reviewer: ```suggestion
When a pod's status is `CrashLoopBackOff` and we run the cluster recovery, the process will not complete. The workaround is to delete the crashing pods manually. The recovery process will then continue.
```

### Long cluster names cause routes to be rejected  (RED25871)

> A cluster name longer than 20 characters will result in a rejected route configuration as the host part of the domain name exceeds 63 characters. The workaround is to limit cluster name to 20 characters or less.

Reviewer: ```suggestion
A cluster name longer than 20 characters will result in a rejected route configuration because the host part of the domain name will exceed 63 characters. The workaround is to limit cluster name to 20 characters or fewer.
```

> ### Cluster CR (REC) errors are not reported after invalid updates (RED25542)

Reviewer: @alexmilowski REC?

A cluster CR specification error is not reported if two or more invalid CR resources are updated in sequence.

### An unreachable cluster has status running (RED32805)

> When a cluster is in an unreachable state the state is still running instead of being reported as an error.

Reviewer: ```suggestion
When a cluster is in an unreachable state, the state is still `running` instead of being reported as an error.
```

### Readiness probe incorrect on failures (RED39300)

> STS Readiness probe doesn't mark a node as not ready when rladmin status on the node fails.

Reviewer: ```suggestion
STS Readiness probe does not mark a node as "not ready" when running `rladmin status` on the node fails.
```

Reviewer: @alexmilowski Should "Readiness" be capitalized here?

### Role missing on replica sets (RED39002)

> The redis-enterprise-operator role is missing permission on replica sets.

Reviewer: ```suggestion
The `redis-enterprise-operator` role is missing permission on replica sets.
```

Reviewer: @alexmilowski I don't really understand this sentence.

### Private registries are not supported on OpenShift 3.11 (RED38579)

> Openshift 3.11 doesn't support dockerhub private registry. This is a known OpenShift issue.

Reviewer: ```suggestion
Openshift 3.11 does not support DockerHub private registries. This is a known OpenShift issue.
```

Reviewer: @alexmilowski Can you hyperlink to the relevant OpenShift issue, if it exists?

### Internal DNS and Kubernetes DNS may have conflicts (RED37462)

> DNS conflicts are possible between the cluster mdns_server and the K8s DNS. This only impacts DNS resolution from within cluster nodes for Kubernetes DNS names.

Reviewer: @alexmilowski Can you elaborate on what is NOT impacted by this conflict?

Author: This has been in every release note that has been published for quite awhile. People understand what this means.

### 5.4.10 negatively impacts 5.4.6 (RED37233)

> Kubernetes-based 5.4.10 deployments seem to negatively impact existing 5.4.6 deployments that share a Kubernetes cluster.

Reviewer: @alexmilowski What does "negatively impact" mean? This is really vague.

Author: This is an old issue that I wish we would remove and the answer is very compilcated.

### Node CPU usage is reported instead of pod CPU usage (RED36884)

In Kubernetes, the node CPU usage we report on is the usage of the Kubernetes worker node hosting the REC pod.

### Clusters must be named "rec" in OLM-based deployments (RED39825)

In OLM-deployed operators, the deployment of the cluster will fail if the name is not "rec". When the operator is deployed via the OLM, the security context constraints (scc) is bound to a specific service account name (i.e., "rec"). The workaround is to name the cluster "rec".

### Master pod label in Rancher (RED42896)

> Master pod is not always labeled in Rancher.

Reviewer: ```suggestion
The master pod is not always labeled in Rancher.
```

> ### Cluster fail to start for clusters with unsynchronized clocks (RED47254)

Reviewer: ```suggestion
### REC clusters fail to start on Kubernetes clusters with unsynchronized clocks (RED47254)
```

Reviewer: @alexmilowski Please confirm that the meaning is still correct here.

> When REC clusters are deployed on clusters with unsynchronized clocks, the cluster does not start correctly. The fix is to use NTP to synchronize the underlying K8s nodes.

Reviewer: ```suggestion
When REC clusters are deployed on Kubernetes clusters with unsynchronized clocks, the REC cluster does not start correctly. The fix is to use NTP to synchronize the underlying K8s nodes.
```

### 5.4.10 negatively impacts 5.4.6 (RED37233)

Kubernetes-based 5.4.10 deployments seem to negatively impact existing 5.4.6 deployments that share a Kubernetes cluster.

### Node CPU usage is reported instead of pod CPU usage (RED36884)

In Kubernetes, the node CPU usage we report on is the usage of the Kubernetes worker node hosting the REC pod.

### Clusters must be named "rec" in OLM-based deployments (RED39825)

In OLM-deployed operators, the deployment of the cluster will fail if the name is not "rec". When the operator is deployed via the OLM, the security context constraints (scc) is bound to a specific service account name (i.e., "rec"). The workaround is to name the cluster "rec".

### Master pod label in Rancher (RED42896)

Master pod is not always labeled in Rancher.

### Cluster fail to start for clusters with unsynchronized clocks (RED47254)

When REC clusters are deployed on clusters with unsynchronized clocks, the cluster does not start correctly. The fix is to use NTP to synchronize the underlying K8s nodes.

> ## Compatibility Notes

Reviewer 2: Maybe omit?

Author: Omit the last issue or compatibility notes (since there are none)?

Author: Removed.
